android-sdk/src/main/java/ai/evolv/android_sdk/EvolvClientImpl.java

```diff
@@ -194,50 +178,228 @@ public void invoke(Object object) {
 
     @Override
     public void confirm() {
+
+        waitForIt.waitFor(evolvContext, EFFECTIVE_GENOME_UPDATED, new EvolvInvocation() {
+            @Override
+            public void invoke(Object value) {
+                JsonObject remoteContext = ((EvolvContextImpl) evolvContext).getRemoteContext();
+                JsonElement allocations = JsonNull.INSTANCE;
+                if (remoteContext.has("experiments")) {
+                    allocations = remoteContext.get("experiments");
+                }
+
+                if (allocations.equals(JsonNull.INSTANCE)
+                        || evolvStore.config.size() == 0
+                        || allocations.getAsJsonObject().size() == 0) {
+                    return;
+                }
+
+                JsonArray entryPointEids = evolvStore.activeEntryPoints();
+
+                if (entryPointEids.size() == 0) {
+                    return;
+                }
+
+                JsonObject confirmations = new JsonObject();
+                if (remoteContext.has("confirmations")) {
+                    confirmations = remoteContext.get("confirmations").getAsJsonObject();
+                }
+
+                JsonObject confirmedCids = new JsonObject();
+                for (Map.Entry<String, JsonElement> entry : confirmations.entrySet() ) {
+                    confirmedCids = entry.getValue().getAsJsonObject().get("cid").getAsJsonObject();
+                }
+
+                JsonObject contaminations = new JsonObject();
+                if (remoteContext.has("contaminations")) {
+                    confirmations = remoteContext.get("contaminations").getAsJsonObject();
```

Comment:
getting contaminations, saving as confirmations

Reply:
thanks for noticing, I corrected that last Wednesday